### PR TITLE
Various fixes to solver class.

### PR DESCRIPTION
**Pull request by jorgensd**
- Add docs and some questions.
- Update images in workflows.
- Update webpage links

@emmetfrancis, @finsberg I've added several notes in the `smartSNESProblem` docstring on input I think we should revise. Any comments on this are very welcome.

### CODE REVIEW

File: smart/solvers.py [hunk 1]
```diff
@@ -493,29 +550,52 @@ def init_petsc_vector(self, j, assemble=False):
...
+                of the domain in the `k`th entry.
+        """
         if k is None:
             return (
                 f"J{i}{j} = dF[{self.active_compartment_names[i]}]"
                 f"/du[{self.active_compartment_names[j]}]"
             )
         else:
+            ij = i * self.dim + j
+            if (num_entries := len(self.Jforms_all[ij])) <= k:
```
Comment (finsberg) [inline smart/solvers.py:570]: Nice usage of the walrus operator `:=`. Note that is a python3.8 feature, which means that we are implicitly dropping support for python3.7.

**Review by finsberg — APPROVED** [review]
I think this looks very good @jorgensd

File: smart/solvers.py (hunk 1 above)
Comment (finsberg) [reply smart/solvers.py:570]: Maybe we should add this constraint to the `pyproject.toml`, i.e https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#requires-python ?

Comment (jorgensd) [reply smart/solvers.py:570]: > Nice usage of the walrus operator `:=`. Note that is a python3.8 feature, which means that we are implicitly dropping support for python3.7.

3.7 is eol June/July, so i think that would be ok

Comment (finsberg) [reply smart/solvers.py:570]: Agree👍

**Review by emmetfrancis — APPROVED** [review]
These changes look great to me! Regarding the MPI communicator used in "solvers", I like the strategy of pulling the communicator from the mesh. I'll work from this for my "multi-mesh MPI" implementation which mostly seems to work now.

**Pull request merged**